### PR TITLE
Add adaption example to Auto_FL

### PR DESCRIPTION
**Pull request by ZiyueXu77**
Fixes # .

### Description

Add a section and example to show how the existing concept of auto_fl can be adopted to a new task and execution environment

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: research/auto-fl-research/README.md [hunk 1]
```diff
@@ -34,6 +34,145 @@ It is designed to combine:
 - `skills/autofl-nvflare-report/` — post-run reporting skill for stopped experiment branches
 - `ACKNOWLEDGEMENTS.md` — provenance and attribution notes
 
+## Local Single-GPU VLM Profile
```
Comment (holgerroth) [inline research/auto-fl-research/README.md]: Let's keep these instructions in auto-fl-research/vlm_local/README.md

In the auto-fl-research/README.md, we should just add a new section towards the and on "How to adapt Auto-FL to new datasets and tasks" and then link to the vlm_local as an example.

**Review by holgerroth — CHANGES_REQUESTED** [review]

File: research/auto-fl-research/README.md (hunk 1 above)
Comment (ZiyueXu77) [reply research/auto-fl-research/README.md]: updated

File: research/auto-fl-research/README.md [hunk 2]
```diff
@@ -11,13 +11,15 @@ effort level: max.
...
-- the **`program.md`-centered control-plane** style popularized by [karpathy/autoresearch](https://github.com/karpathy/autoresearch), where the human primarily evolves the research instructions and the agent iterates on a bounded code surface.
+- task-specific training profiles that preserve **diff-based uploads** and comparable cross-site scoring; and
+- the **`program.md` control-plane plus task profile** style popularized by [karpathy/autoresearch](https://github.com/karpathy/autoresearch), where the human primarily evolves the research instructions and the agent iterates on a bounded code surface.
 
 ## What is included
 
-- `program.md` — the main agent control plane
-- `AGENTS.md`, `CLAUDE.md` — thin repository guardrails that point back to `program.md`
+- `program.md` — the general agent control plane
+- `cifar10.md` — the default CIFAR-10/H100 task profile
```
Comment (holgerroth) [inline research/auto-fl-research/README.md]: Let's move cifar10.md and vlm_med.md into a "tasks" folder so we can expand in the future.

File: research/auto-fl-research/README.md [hunk 3]
```diff
@@ -174,7 +176,7 @@ export PYTHON=.venv/bin/python
 Treat that PYTHON value as authoritative. First verify it with `test -x "$PYTHON"` and `"$PYTHON" -c "import sys; assert sys.version_info[:2] == (3, 12), sys.version; print(sys.executable)"`, then use that exact interpreter for validation, smoke tests, candidate runs, plotting, summaries, and reports.
 Do not create a virtual environment, install dependencies, or search for alternate Python interpreters unless I explicitly ask you to. If `.venv/bin/python` is missing, invalid, or not Python 3.12, stop and tell me to rerun the README preflight in this directory with `python3.12`.
 
-Use the default H100 candidate budget unless `program.md` says otherwise:
+Use the default H100 candidate budget unless `cifar10.md` says otherwise:
```
Comment (holgerroth) [inline research/auto-fl-research/README.md]: "unless task profile says otherwise"?

File: research/auto-fl-research/README.md [hunk 4]
```diff
@@ -189,7 +191,22 @@ After every reviewed batch, run `"${PYTHON}" scripts/plateau_watchdog.py results
 Commit `results.tsv` locally after the baseline and after each reviewed batch. Commit surviving code changes locally on the active `autoresearch/` branch as soon as they are kept; do not let kept mutations accumulate only in the working tree. Do not require pushing from inside the devcontainer.
 ```
 
-## Scoring recommendation
+To override the default task profile, say so explicitly in the first prompt.
```
Comment (holgerroth) [inline research/auto-fl-research/README.md]: I don't think we need this section. It's repeated below in "How to adapt Auto-FL to new datasets and tasks"

File: research/auto-fl-research/README.md [hunk 5]
```diff
@@ -278,6 +295,48 @@ To generate an autoresearch-style progress image from the ledger:
...
+
+- a profile file such as `cifar10.md` or `vlm_med.md` for the task contract,
+  fixed comparison budget, metric, environment assumptions, and preferred edit
+  surface.
+- `mutation_schema.yaml` entries for bounded mutation axes that the agent may
+  choose during a campaign.
+- Task-specific `client.py`, `job.py`, `model.py`, `train_utils.py`, and
+  `data/` files when the dataset, training loop, model state, or score
+  extraction differs from the default profile.
+- `requirements.txt` only for dependencies that are specific to the new task
```
Comment (holgerroth) [inline research/auto-fl-research/README.md]: likely each tasks would need its own subfolder for profile.md and requirements.txt

File: research/auto-fl-research/README.md (hunk 4 above)
Comment (ZiyueXu77) [reply research/auto-fl-research/README.md]: removed

File: research/auto-fl-research/README.md (hunk 2 above)
Comment (ZiyueXu77) [reply research/auto-fl-research/README.md]: moved

Comment (ZiyueXu77) [reply research/auto-fl-research/README.md]: moved

File: research/auto-fl-research/README.md (hunk 3 above)
Comment (ZiyueXu77) [reply research/auto-fl-research/README.md]: updated

File: research/auto-fl-research/README.md (hunk 5 above)
Comment (ZiyueXu77) [reply research/auto-fl-research/README.md]: structured

**Review by holgerroth — DISMISSED** [review]
Task profiles are a good addition!

File: research/auto-fl-research/tasks/vlm_med/client.py [hunk 6]
```diff
@@ -0,0 +1,934 @@
...
+
+def _load_adapter_state_into_vlm(adapter_model, vlm_model):
+    from peft.utils.save_and_load import set_peft_model_state_dict
+
+    peft_state = adapter_state_to_peft_state(adapter_model.state_dict())
+    result = set_peft_model_state_dict(vlm_model, peft_state)
+    unexpected = getattr(result, "unexpected_keys", None)
+    if unexpected:
+        raise ValueError(f"Unexpected VLM adapter keys while loading global state: {unexpected[:4]}")
+
```
[the remark this answers is not in the document]
Comment (ZiyueXu77) [reply research/auto-fl-research/tasks/vlm_med/client.py:458]: fixed

**Review by holgerroth — APPROVED** [review]
